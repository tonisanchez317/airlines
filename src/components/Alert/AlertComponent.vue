<script>
import includes from 'lodash/includes';
import ALERT_TYPES from './types';

const DEFAULT_ALERT_TYPE = ALERT_TYPES.primary;

export default {
  name: 'AlertComponent',

  props: {
    type: {
      type: String,
      default: DEFAULT_ALERT_TYPE,
    },
  },

  computed: {
    validType() {
      return (includes(ALERT_TYPES, this.type) && this.type) || DEFAULT_ALERT_TYPE;
    },
  },
};
</script>

<template>
  <div
    :class="`alert alert-${validType} alert-dismissible fade show`"
    role="alert"
  >
    <slot />
    <button
      type="button"
      class="close"
      data-dismiss="alert"
      aria-label="Close"
      @click="$emit('on-close')"
    >
      <span aria-hidden="true">&times;</span>
    </button>
  </div>
</template>
