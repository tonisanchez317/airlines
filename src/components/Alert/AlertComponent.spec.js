import { shallowMount } from '@vue/test-utils';
import AlertComponent from './AlertComponent.vue';
import ALERT_TYPES from './types';

const alertTypes = [
  ...Object.values(ALERT_TYPES),
  'test',
];

describe('Given AlertComponent', () => {
  alertTypes.forEach(alertType => {
    describe(`When component receives alert type "${alertType}"`, () => {
      it('Then HTML rendered should be the same than original snapshot', () => {
        const wrapper = shallowMount(AlertComponent, {
          propsData: {
            type: alertType,
          },
        });

        expect(wrapper).toMatchSnapshot();
      });
    });
  });
});
