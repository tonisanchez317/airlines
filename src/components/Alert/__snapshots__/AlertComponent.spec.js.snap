// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given AlertComponent When component receives alert type "danger" Then HTML rendered should be the same than original snapshot 1`] = `<div role="alert" class="alert alert-danger alert-dismissible fade show"> <button type="button" data-dismiss="alert" aria-label="Close" class="close"><span aria-hidden="true">×</span></button></div>`;

exports[`Given AlertComponent When component receives alert type "dark" Then HTML rendered should be the same than original snapshot 1`] = `<div role="alert" class="alert alert-dark alert-dismissible fade show"> <button type="button" data-dismiss="alert" aria-label="Close" class="close"><span aria-hidden="true">×</span></button></div>`;

exports[`Given AlertComponent When component receives alert type "info" Then HTML rendered should be the same than original snapshot 1`] = `<div role="alert" class="alert alert-info alert-dismissible fade show"> <button type="button" data-dismiss="alert" aria-label="Close" class="close"><span aria-hidden="true">×</span></button></div>`;

exports[`Given AlertComponent When component receives alert type "light" Then HTML rendered should be the same than original snapshot 1`] = `<div role="alert" class="alert alert-light alert-dismissible fade show"> <button type="button" data-dismiss="alert" aria-label="Close" class="close"><span aria-hidden="true">×</span></button></div>`;

exports[`Given AlertComponent When component receives alert type "primary" Then HTML rendered should be the same than original snapshot 1`] = `<div role="alert" class="alert alert-primary alert-dismissible fade show"> <button type="button" data-dismiss="alert" aria-label="Close" class="close"><span aria-hidden="true">×</span></button></div>`;

exports[`Given AlertComponent When component receives alert type "secondary" Then HTML rendered should be the same than original snapshot 1`] = `<div role="alert" class="alert alert-secondary alert-dismissible fade show"> <button type="button" data-dismiss="alert" aria-label="Close" class="close"><span aria-hidden="true">×</span></button></div>`;

exports[`Given AlertComponent When component receives alert type "success" Then HTML rendered should be the same than original snapshot 1`] = `<div role="alert" class="alert alert-success alert-dismissible fade show"> <button type="button" data-dismiss="alert" aria-label="Close" class="close"><span aria-hidden="true">×</span></button></div>`;

exports[`Given AlertComponent When component receives alert type "test" Then HTML rendered should be the same than original snapshot 1`] = `<div role="alert" class="alert alert-primary alert-dismissible fade show"> <button type="button" data-dismiss="alert" aria-label="Close" class="close"><span aria-hidden="true">×</span></button></div>`;

exports[`Given AlertComponent When component receives alert type "warning" Then HTML rendered should be the same than original snapshot 1`] = `<div role="alert" class="alert alert-warning alert-dismissible fade show"> <button type="button" data-dismiss="alert" aria-label="Close" class="close"><span aria-hidden="true">×</span></button></div>`;
